lib/compiler-singlepass/src/codegen_x64.rs

```diff
@@ -1423,7 +1421,7 @@ impl<'a> FuncGen<'a> {
         if memarg.offset != 0 {
             self.assembler.emit_add(
                 Size::S32,
-                Location::Imm32(memarg.offset),
+                Location::Imm32(memarg.offset as u32), // we don’t support 64-bit memory, and this module was validated
```

Comment:
I’d say `u32::try_from(...).unwrap()` would still be a more future-proof thing to do, but I don't have strong feelings here.